source/blog/2017-06-06-using-kubeadm-with-cri-o.html.md

```diff
@@ -0,0 +1,312 @@
+[certificates] Generated API server certificate and key.
+[certificates] API Server serving cert is signed for DNS names [kubeadm-master kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.138.0.4]
+[certificates] Generated API server kubelet client certificate and key.
+[certificates] Generated service account token signing key and public key.
+[certificates] Generated front-proxy CA certificate and key.
+[certificates] Generated front-proxy client certificate and key.
+[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
+[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
+[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
+[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
+[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
+[apiclient] Created API client, waiting for the control plane to become ready
+[apiclient] All control plane components are healthy after 16.772251 seconds
+[apiclient] Waiting for at least one node to register and become ready
+[apiclient] First node is ready after 5.002536 seconds
+[apiclient] Test deployment succeeded
+[token] Using token: <token>
+[apiconfig] Created RBAC rules
+[addons] Created essential addon: kube-proxy
+[addons] Created essential addon: kube-dns
+
+Your Kubernetes master has initialized successfully!
+
+To start using your cluster, you need to run (as a regular user):
+
+  sudo cp /etc/kubernetes/admin.conf $HOME/
+  sudo chown $(id -u):$(id -g) $HOME/admin.conf
+  export KUBECONFIG=$HOME/admin.conf
+
+You should now deploy a pod network to the cluster.
+Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
+  http://kubernetes.io/docs/admin/addons/
+
+You can now join any number of machines by running the following on each node
+as root:
+
+  kubeadm join --token 6b7a29.95a2995f65e1d3c9 192.168.122.34:6443
+```
+
+Make a record of the `kubeadm join` command that `kubeadm init` outputs. You will need this in a moment.
+
+Let's explain the flags you used:
+
+- `--pod-network-cidr=10.244.0.0/16`: this is required as you're going to use Flannel to provide network for the cluster
+- `--skip-preflight-checks`: `kubeadm` still assumes docker as the only container runtime which can be used with Kubernetes. You need this flag in order to silence `kubeadm init` when it warns us that docker is not installed.  We've  opened an issue upstream to take care of this at https://github.com/kubernetes/kubeadm/issues/285.
+
+ The master is now up and running. Only thing remaining to do on the master deploying a network add-on for the cluster:
+
+```sh
+fedora.vm $ export KUBECONFIG=/etc/kubernetes/admin.conf
+fedora.vm $ kubectl apply -f https://gist.githubusercontent.com/sameo/cf92f65ae54a87807ed294f3de658bcf/raw/95d9a66a2268b779dbb25988541136d1ed2fbfe2/flannel.yaml
+serviceaccount "flannel" created
```

Comment:
just verified it works thanks to https://github.com/kubernetes-incubator/cri-o/pull/580